### PR TITLE
Add seccomp support for PARISC (HPPA)

### PR DESCRIPTION
**Pull request by thesamesam**
Fixes: #23180

### CODE REVIEW

File: src/shared/seccomp-util.c [hunk 1]
```diff
@@ -132,6 +137,14 @@ const char* seccomp_arch_to_string(uint32_t c) {
                 return "mips64-le";
         case SCMP_ARCH_MIPSEL64N32:
                 return "mips64-le-n32";
+#ifdef SCMP_ARCH_PARISC
```
Comment (thesamesam) [inline src/shared/seccomp-util.c:140]: I've opted to keep the #ifdef guards but they're trivial to remove. Support for HPPA landed in libseccomp in 2016 but was released in libseccomp 2.4.0 (2019).

Comment (yuwata) [reply src/shared/seccomp-util.c:140]: We have similar ifdef for `SCMP_ARCH_RISCV64`. So, that's OK. But please add similar `#ifdef` at all places where `SCMP_ARCH_PARISC` or friends are used.

File: src/shared/seccomp-util.c [hunk 2]
```diff
@@ -80,6 +80,11 @@ uint32_t seccomp_local_archs[] = {
                 SCMP_ARCH_MIPSEL64,
                 SCMP_ARCH_MIPS64N32,
                 SCMP_ARCH_MIPSEL64N32, /* native */
+#elif defined(__hppa64__)
+                SCMP_ARCH_PARISC,
+                SCMP_ARCH_PARISC64,    /* native */
+#elif defined(__hppa__)
+                SCMP_ARCH_PARISC,
```
Comment (yuwata) [inline src/shared/seccomp-util.c]: Please add `&& defined (SCMP_ARCH_PARISC)` or similar here.

**Review by yuwata — CHANGES_REQUESTED** [review]
Looks mostly good.

**Comment by yuwata** [comment]
BTW, please read http://systemd.io/PORTING_TO_NEW_ARCHITECTURES/, and if you'd like, please also update them.
1 is already addressed, 4 and 5 are addressed by this PR. I am not sure about the other items.

**Comment by yuwata** [comment]
Also, please drop the Signed-off-by from the commit message. We do not use it.

**Comment by thesamesam** [comment]
Thanks for the feedback. I'll get on it soon!

**Comment by yuwata** [comment]
Ping?

**Comment by thesamesam** [comment]
Let me finish this off. I'll take a look tonight. I was sort of waiting to see if I could get away without the exec stack stuff but while there's been kernel changes, not seen anything else there yet, so..

**Review by yuwata — APPROVED** [review]

**Comment by thesamesam** [comment]
Two failures are unrelated (timeouts). focal-arm64 seems stuck.

**Pull request merged**